lib/prawn/icon/interface.rb

```diff
@@ -225,7 +225,17 @@ def format_hash
 
     def render
       @pdf.font(@data.path) do
-        @pdf.text @unicode, @options
+        opts = @options.dup
```

Comment:
Are we able to remove the shallow copy by avoiding destructive operations to the `@options` hash?

For example, perhaps something like the following:

```ruby
size_mode = @options.fetch(:size_mode, :font_size)
requested_size = @options.fetch(:size) { @pdf.font_size }
adjusted_size = requested_size

if size_mode == :icon_size
  ...
  adjusted_size = requested_size / actual_height * requested_size
end

@pdf.text @unicode, @options.merge(size: adjusted_size)
```

Reply:
If it's ok to pass the `:size_mode` key on to Prawn then your suggestion is fine by me. I'll update the PR.

Reply:
Good point - I think that's acceptable here (Prawn won't do anything with the extra param).

Reply:
I've shuffled the code around a bit so that you only pay the Hash copy cost in the `:icon_height` case. In your suggestion we were still always copying in the `merge` call.